$OpenBSD$

Don't deadlock if nanosleep is interrupted.
Taken from a larger pull request upstream.

Index: hhpc.c
--- hhpc.c.orig
+++ hhpc.c
@@ -35,6 +35,7 @@
 #include <time.h>
 #include <getopt.h>
 
+#include <errno.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
@@ -90,8 +91,11 @@ static void delay(time_t sec, long msec) {
     sleep.tv_sec  = sec;
     sleep.tv_nsec = (msec % 1000) * 1000 * 1000;
 
-    if (nanosleep(&sleep, NULL) == -1) {
-        signalHandler(0);
+    while (nanosleep(&sleep, &sleep) == -1) {
+	if (errno == EINTR)
+		continue;
+	else
+ 	       signalHandler(0);
     }
 }
 
